### ISSUE TITLE
No genotype output from vcf2snapp.py

### ISSUE THREAD

**Issue by pbpearman** [CoBiG2/RAD_Tools#1]
HI. I am trying vcf2snapp.py for the obvious reason.  My test vcf has 6 individuals and about 1800 loci.  The output I get from the script only shows the individual ids:

#NEXUS
BEGIN Data;
	DIMENSIONS NTAX=6 NCHAR=0;
	FORMAT DATATYPE=integer INTERLEAVE=no missing=?;
Matrix
AC1134	
AC1186	
AC1234	
AC1425	
AC1440	
AC3928	
;
END;

I think this may be because my vcf includes a field for depth, DP, in the genotype format.  Here is part of the vcf header and first data line:

##fileformat=VCFv4.0
##FILTER=<ID=PASS,Description="All filters passed">
##fileDate=Tue Jul  3 09:49:07 2018
##source=GBS-SNP-CROP
##phasing=partial
##INFO=<ID=AC,Number=1,Type=Integer,Description=Allele Count>
##INFO=<ID=AF,Number=1,Type=Integer,Description=Allele Frequency>
##INFO=<ID=DP,Number=1,Type=Integer,Description=Combined Depth>
##INFO=<ID=NS,Number=1,Type=Integer,Description=Number of Samples With Data>
##FORMAT=<ID=GT,Number=1,Type=String,Description=Genotype>
##FORMAT=<ID=DP,Number=1,Type=Integer,Description=Read Depth>
##contig=<ID=MockRefGenome>
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	AC1134	AC1186	AC1234	AC1425	AC1440	AC3928
MockRefGenome	251784	.	A	G	40	PASS	.	GT:DP	0/1:8	0/0:9	0/0:17	0/0:10	0/0:8	0/0:6

My python is primitive, but I think these lines in the script need to be modified a little to work on my vcf:
            # Record data for each Taxon
            for tx in nexus_data:
                # Get genotype
                gen = fields[taxa_list.index(tx)]

                if gen == "./.":
                    nexus_data[tx].append("?")
                elif gen == "0/0":
                    nexus_data[tx].append("0")
                elif gen == "1/1":
                    nexus_data[tx].append("2")
                elif gen == "1/0" or gen == "0/1":
                    nexus_data[tx].append("1")

But this doesn't produce the desired .nex file, perhaps because each genotype has the depth field.  I'm not sure. I'm not very capable in python.
Could you suggest a fix so that vcf2snapp.py will work with my vcf file?  Any help would be greatly appreciated.  I've attached a .zip archive of a complete vcf file
[Eu_SNPs_2000_100pct_covered_1.recode.vcf.zip](https://github.com/CoBiG2/RAD_Tools/files/4509747/Eu_SNPs_2000_100pct_covered_1.recode.vcf.zip)

**Comment by StuntsPT** [comment]
Hi @pbpearman,
I have uploaded a new version of `vcf2snapp.py` which should address your issue.
Your assessement was correct and the lines you mentioned did need fixing. But there were other issues too, like the parser having been built for VCFs without chromossome information (which is now fixed, btw).

You now have to run it using the option `-chr`, so something like this:

```bash
python vcf2snappy.py -in Eu_SNPs_2000_100pct_covered_1.recode.vcf -chr
```
Let me know if it works for you.
